### ISSUE TITLE
Padding impacts parallel_logits computation, affecting PPO logprobs 

### ISSUE THREAD

**Issue by shengyangs** [NVIDIA/NeMo-Aligner#68]
**Describe the bug**

When the input tokens are padded in a batch, the computed parallel_logits are different with that without padding. Consequently , the log probs are different. Specifically, in PPO, the `prev_logprobs` are computed without padding ([link](https://github.com/NVIDIA/NeMo-Aligner/blob/main/nemo_aligner/models/nlp/gpt/megatron_gpt_ppo_actor.py#L122)) while the `cur_logprobs` are computed with padding ([link](https://github.com/NVIDIA/NeMo-Aligner/blob/main/nemo_aligner/models/nlp/gpt/megatron_gpt_ppo_actor.py#L224)). Even with the same input tokens and the same model parameters, `cur_logprobs` and `prev_logprobs` are different. 

**Steps/Code to reproduce bug**

1. At step=0, the `ppo_ratio` should be `1` since the model is not updated yet. However, if we check the wandb plots, the `ppo_ratio` is not `1` even at the first step.

2. We can reproduce this by setting `global_batch_size=2` and `num_rollouts=2`. 

3. The figure below shows that truncating the input tokens till `T=100` actually affects the outputs at `T=2` (comparing the 2nd row of the two inputs). This validates that padding affects outputs.
 
![Screenshot 2023-12-27 at 3 37 23 PM](https://github.com/NVIDIA/NeMo-Aligner/assets/150360624/9865b3ff-0616-41a4-b9d0-e82cdfa5451a)

**Comment by shengyangs** [comment]
@gshennvm @odelalleau @HeyyyyyyG for PPO viz.
